scalableminds/webknossos
# Allow merge and min-cut when rightclicking on voxel (instead of node)

### URL of deployed dev instance (used for testing):
- [https://___.webknossos.xyzhttps://proofreadingqolsplitandmerge.webknossos.xyz/](https://proofreadingqolsplitandmerge.webknossos.xyz/)

### Steps to test:
- create annotation for test-agglomerate-file DS
- select hdf5 mapping
- select proofreading tool
- click on segment to load agglomerate skeleton
- merging
  - select a node
  - rightclick on another segment (not a node) and select "merge with active segment"
    - --> segments should get merged (agglomerate skeleton will be reloaded)

~- splitting~
  ~- select a node~
    ~- rightclick on the same segment somewhere where it makes sense* and select "split from active segment"~
    ~- segments should get split~
- min-cutting
  - select a node
    - rightclick on the same segment somewhere where it makes sense** and select "min-cut from active segment"
    - segments should get split

Footnotes:
~\* for the user it's virtually impossible to know which voxel position would work for splitting, since one cannot see the oversegmentation (a toast will appear which tells you about this, though). if one clicks in the vicinity of another node of the same agglomerate skeleton, the chances are good that the split will succeed. however, it still might be the case that one clicks on the same over-segment. even if the split works, the user might get confused because nothing visibly happens. this is the case, when there are multiple edges which need to be split (however, the user action would only remove one edge).~
** for min-cutting there is the same risk of rightclicking the very same oversegment. however, if the target position refers to another over-segment, the split should succeed regardless of how many (indirect) edges exist, because they will be deleted by the min-cut.

the above problems might be only of theoretical nature. when working with actual segmentations and mergers/splits (and not dummy ones), the problem might be negligible. if these still appear, wk will recommend to use the agglomerate structure to perform the split or min-cut.

### Issues:
- fixes #6426

------
(Please delete unneeded items, merge only when none are left open)
